<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>SettingDialog</class>
 <widget class="QDialog" name="SettingDialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>450</width>
    <height>329</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Settings</string>
  </property>
  <layout class="QGridLayout" name="gridLayout_2">
   <item row="0" column="0">
    <layout class="QGridLayout" name="gridLayout">
     <item row="0" column="0">
      <widget class="QGroupBox" name="previewGroupBox">
       <property name="title">
        <string>Preview</string>
       </property>
       <layout class="QFormLayout" name="formLayout">
        <property name="margin">
         <number>8</number>
        </property>
        <item row="0" column="0">
         <widget class="QLabel" name="maxImageWidthLabel">
          <property name="text">
           <string>Max image width</string>
          </property>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="QLineEdit" name="maxImageWidthLineEdit"/>
        </item>
        <item row="1" column="0">
         <widget class="QLabel" name="maxImageHeightLabel">
          <property name="text">
           <string>Max image height</string>
          </property>
         </widget>
        </item>
        <item row="1" column="1">
         <widget class="QLineEdit" name="maxImageHeightLineEdit"/>
        </item>
        <item row="2" column="0">
         <widget class="QLabel" name="defaultImageLabel">
          <property name="text">
           <string>Default image</string>
          </property>
         </widget>
        </item>
        <item row="2" column="1">
         <widget class="QLineEdit" name="defaultImageLineEdit"/>
        </item>
       </layout>
      </widget>
     </item>
     <item row="0" column="1">
      <widget class="QGroupBox" name="buffer2dGroupBox">
       <property name="title">
        <string>Buffer 2D kernels</string>
       </property>
       <layout class="QFormLayout" name="formLayout_3">
        <property name="margin">
         <number>8</number>
        </property>
        <item row="0" column="0">
         <widget class="QLabel" name="erodeKernelLabel">
          <property name="text">
           <string>Erode</string>
          </property>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="QComboBox" name="erodeKernelComboBox">
          <item>
           <property name="text">
            <string>erode</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>erode_c4</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>erode_c4_pragma</string>
           </property>
          </item>
         </widget>
        </item>
        <item row="1" column="0">
         <widget class="QLabel" name="dilateKernelLabel">
          <property name="text">
           <string>Dilate</string>
          </property>
         </widget>
        </item>
        <item row="1" column="1">
         <widget class="QComboBox" name="dilateKernelComboBox">
          <item>
           <property name="text">
            <string>dilate</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>dilate_c4</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>dilate_c4_pragma</string>
           </property>
          </item>
         </widget>
        </item>
        <item row="2" column="0">
         <widget class="QLabel" name="gradientKernelLabel">
          <property name="text">
           <string>Gradient       </string>
          </property>
         </widget>
        </item>
        <item row="2" column="1">
         <widget class="QComboBox" name="gradientKernelComboBox">
          <item>
           <property name="text">
            <string>gradient</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>gradient_c4</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>gradient_c4_pragma</string>
           </property>
          </item>
         </widget>
        </item>
       </layout>
      </widget>
     </item>
     <item row="1" column="0">
      <widget class="QGroupBox" name="openclGroupBox">
       <property name="title">
        <string>OpenCL</string>
       </property>
       <layout class="QFormLayout" name="formLayout_2">
        <property name="margin">
         <number>8</number>
        </property>
        <item row="0" column="0">
         <widget class="QLabel" name="useAtomicCountersLabel">
          <property name="text">
           <string>Use atomic counters</string>
          </property>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="QCheckBox" name="useAtomicCountersCheckBox"/>
        </item>
        <item row="1" column="0">
         <widget class="QLabel" name="openGInteropLabel">
          <property name="text">
           <string>OpenGL Interop</string>
          </property>
         </widget>
        </item>
        <item row="1" column="1">
         <widget class="QCheckBox" name="openGInteropCheckBox"/>
        </item>
        <item row="2" column="0">
         <widget class="QLabel" name="datatypeLabel">
          <property name="text">
           <string>Datatype</string>
          </property>
         </widget>
        </item>
        <item row="2" column="1">
         <widget class="QComboBox" name="datatypeComboBox">
          <item>
           <property name="text">
            <string>uchar</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>uint</string>
           </property>
          </item>
         </widget>
        </item>
        <item row="3" column="0">
         <widget class="QLabel" name="workgroupSizeXLabel">
          <property name="text">
           <string>Workgroup size X</string>
          </property>
         </widget>
        </item>
        <item row="3" column="1">
         <widget class="QComboBox" name="workgroupSizeXComboBox">
          <item>
           <property name="text">
            <string>1</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>2</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>4</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>8</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>16</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>32</string>
           </property>
          </item>
         </widget>
        </item>
        <item row="4" column="0">
         <widget class="QLabel" name="workgroupSizeYLabel">
          <property name="text">
           <string>Workgroup size Y</string>
          </property>
         </widget>
        </item>
        <item row="4" column="1">
         <widget class="QComboBox" name="workgroupSizeYComboBox">
          <item>
           <property name="text">
            <string>1</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>2</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>4</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>8</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>16</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>32</string>
           </property>
          </item>
         </widget>
        </item>
       </layout>
      </widget>
     </item>
     <item row="1" column="1">
      <widget class="QGroupBox" name="buffer1dGroupBox">
       <property name="title">
        <string>Buffer 1D</string>
       </property>
       <layout class="QFormLayout" name="formLayout_4">
        <property name="margin">
         <number>8</number>
        </property>
        <item row="0" column="0">
         <widget class="QLabel" name="erodeKernelLabel_2">
          <property name="text">
           <string>Erode</string>
          </property>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="QComboBox" name="erodeKernelComboBox_2">
          <item>
           <property name="text">
            <string>erode</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>erode_local</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>erode_c4_local</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>erode_c4_local_unroll</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>erode_c4_local_pragma</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>erode4_local</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>erode4_c4_local</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>erode4_c4_local_unroll</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>erode4_c4_local_pragma</string>
           </property>
          </item>
         </widget>
        </item>
        <item row="1" column="0">
         <widget class="QLabel" name="dilateKernelLabel_2">
          <property name="text">
           <string>Dilate</string>
          </property>
         </widget>
        </item>
        <item row="1" column="1">
         <widget class="QComboBox" name="dilateKernelComboBox_2">
          <item>
           <property name="text">
            <string>dilate</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>dilate_local</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>dilate_c4_local</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>dilate_c4_local_unroll</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>dilate_c4_local_pragma</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>dilate4_local</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>dilate4_c4_local</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>dilate4_c4_local_unroll</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>dilate4_c4_local_pragma</string>
           </property>
          </item>
         </widget>
        </item>
        <item row="2" column="0">
         <widget class="QLabel" name="gradientKernelLabel_2">
          <property name="text">
           <string>Gradient</string>
          </property>
         </widget>
        </item>
        <item row="2" column="1">
         <widget class="QComboBox" name="gradientKernelComboBox_2">
          <item>
           <property name="text">
            <string>gradient</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>gradient_local</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>gradient_c4_local</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>gradient_c4_local_unroll</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>gradient_c4_local_pragma</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>gradient4_local</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>gradient4_c4_local</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>gradient4_c4_local_unroll</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>gradient4_c4_local_pragma</string>
           </property>
          </item>
         </widget>
        </item>
        <item row="3" column="0">
         <widget class="QLabel" name="subtractKernelLabel">
          <property name="text">
           <string>Subtract</string>
          </property>
         </widget>
        </item>
        <item row="3" column="1">
         <widget class="QComboBox" name="subtractKernelComboBox">
          <item>
           <property name="text">
            <string>subtract</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>subtract4</string>
           </property>
          </item>
         </widget>
        </item>
        <item row="4" column="0">
         <widget class="QLabel" name="hitmissMemTypeLabel">
          <property name="text">
           <string>HM memtype</string>
          </property>
         </widget>
        </item>
        <item row="4" column="1">
         <widget class="QComboBox" name="hitmissMemTypeComboBox">
          <item>
           <property name="text">
            <string>local</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>global</string>
           </property>
          </item>
         </widget>
        </item>
       </layout>
      </widget>
     </item>
     <item row="2" column="0" colspan="2">
      <layout class="QHBoxLayout">
       <property name="spacing">
        <number>6</number>
       </property>
       <property name="margin">
        <number>0</number>
       </property>
       <item>
        <spacer>
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>131</width>
           <height>31</height>
          </size>
         </property>
        </spacer>
       </item>
       <item>
        <widget class="QPushButton" name="pbOk">
         <property name="text">
          <string>OK</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="pbCancel">
         <property name="text">
          <string>Cancel</string>
         </property>
        </widget>
       </item>
      </layout>
     </item>
    </layout>
   </item>
   <item row="0" column="1">
    <spacer name="horizontalSpacer">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>40</width>
       <height>20</height>
      </size>
     </property>
    </spacer>
   </item>
   <item row="1" column="0">
    <spacer name="verticalSpacer">
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>20</width>
       <height>40</height>
      </size>
     </property>
    </spacer>
   </item>
  </layout>
 </widget>
 <tabstops>
  <tabstop>maxImageWidthLineEdit</tabstop>
  <tabstop>maxImageHeightLineEdit</tabstop>
  <tabstop>useAtomicCountersCheckBox</tabstop>
  <tabstop>openGInteropCheckBox</tabstop>
  <tabstop>datatypeComboBox</tabstop>
  <tabstop>workgroupSizeXComboBox</tabstop>
  <tabstop>workgroupSizeYComboBox</tabstop>
  <tabstop>erodeKernelComboBox</tabstop>
  <tabstop>dilateKernelComboBox</tabstop>
  <tabstop>gradientKernelComboBox</tabstop>
  <tabstop>erodeKernelComboBox_2</tabstop>
  <tabstop>dilateKernelComboBox_2</tabstop>
  <tabstop>gradientKernelComboBox_2</tabstop>
  <tabstop>subtractKernelComboBox</tabstop>
  <tabstop>hitmissMemTypeComboBox</tabstop>
  <tabstop>pbOk</tabstop>
  <tabstop>pbCancel</tabstop>
 </tabstops>
 <resources/>
 <connections>
  <connection>
   <sender>pbOk</sender>
   <signal>clicked()</signal>
   <receiver>SettingDialog</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>278</x>
     <y>253</y>
    </hint>
    <hint type="destinationlabel">
     <x>96</x>
     <y>254</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pbCancel</sender>
   <signal>clicked()</signal>
   <receiver>SettingDialog</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>369</x>
     <y>253</y>
    </hint>
    <hint type="destinationlabel">
     <x>179</x>
     <y>282</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
